float32[] distance_list
int16[] id_list